src/main/java/com/naijagov/naijagov/controller/NaijagovController.java

```diff
@@ -1,9 +1,28 @@
 package com.naijagov.naijagov.controller;
 
+import com.naijagov.naijagov.model.Form;
+import com.naijagov.naijagov.service.PollingUnitService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.servlet.ModelAndView;
 
 @Controller
 public class NaijagovController{
 
-}
-    
\ No newline at end of file
+    @GetMapping("/pollingunits")
+    public ModelAndView pollingUnits(Pageable pageable, @ModelAttribute Form form, ModelAndView mv){
+        form.setPollingUnits(pollingUnitService.listAllPollingUnits(pageable));
+        mv.setViewName("pollingunits");
+        return mv;
+    }
+    
+    @Autowired
```

Comment:
I think it's best practice to always put your constructor first after the class level fields. That way it's easier to keep track of all important attributes of your class

Reply:
@sfagade Thank you for your observations.

According to  the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents), _there's no single correct recipe for ordering members of a class_ including constructors however we should **follow a logical order** and be able to **articulate our rationale.**

**My rationale:** OOP promotes information-hiding meaning only the relevant behaviour of an object should be exposed.  The key interface of an object are its primary concern rather that the implementation which includes private members and the constructors.  Infact, it's been suggested that calling constructors directly should be discouraged ([Java's new considered harmful](http://www.drdobbs.com/javas-new-considered-harmful/184405016)).  This is also supported by the popular **Effective Java** book ([Consider static factory methods instead of constructors](http://www.informit.com/articles/article.aspx?p=1216151)).

Hence,  my **ordering style** is:

1.  Public before private
2.  Method before constructor
3.  Method before field

Reply:
Point one is well appreciated. However depending on the operations you perform in your constructor and how many methods you have in the class, you might want to consider how easily readable your class will be for the next guy that might inherit your code